src/messages.ts

```diff
@@ -38,6 +38,77 @@ class Key {
  */
 export type LoaderFunction = (locale: string) => Messages;
 
+/**
+ * Different message file content parsers. This allows us to have js, json, and md. Maybe we will eventually support yaml, etc.
+ */
+type FileParser = (filePath: string, fileContents: string) => Map<string, AnyJson>;
+
+const markdownLoader: FileParser = (filePath: string, fileContents: string): Map<string, AnyJson> => {
+  const map = new Map<string, AnyJson>();
+  const hasContent = (lineItem: string) => !/^\s*$/g.exec(lineItem);
+
+  // Filter out sections that don't have content
+  const sections = fileContents.split(/^#\s*/gm).filter(hasContent);
+
+  for (const section of sections) {
+    const lines = section.split('\n');
+    const firstLine = lines.shift();
+    const rest = lines.join('\n').trim();
+
+    if (firstLine && rest.length > 0) {
+      const key = firstLine.trim();
+      const nonEmptyLines = lines.filter((line) => !!line.trim());
+      // If every entry in the value is a list item, then treat this as a list. Indented lines are part of the list.
+      if (nonEmptyLines.every((line) => /^[*-]\s+|^ {2}/.exec(line))) {
+        const listItems = rest.split(/^[*-]\s+/gm).filter(hasContent);
+        const values = listItems.map((item) =>
+          item
+            .split('\n')
+            // new lines are ignored in markdown lists
+            .filter((line) => !!line.trim())
+            // trim off the indentation
+            .map((line) => line.trim())
+            // put it back together
+            .join('\n')
+        );
+        map.set(key, values);
+      } else {
+        map.set(key, rest);
+      }
+    } else {
+      // messages.js should have no internal dependencies.
```

Comment:
I'm not sure this comment applies to this error

Reply:
It does, because you can't have SfdxError. I will make it more clear.